src/api/customer/customer-controller.ts

```diff
@@ -15,6 +15,33 @@ const createCustomer = (): CustomerCreationRequestHandler[] => {
     ]
 }
 
+const listAllCustomers = (): ListAllCustomersRequestHandler [] => {
+    return [
+        listAllCustomersValidator(),
+        paginationSerializer,
+        listAllBaseCustomers,
+        listAllCustomersSerializer,
+        (req, res) => {
+            res.status(OK).json(res.locals.customerToRespond);
```

Comment:
Essa rota também deveria devolver um erro `NOT_FOUND` caso não encontrasse registros.

Pensa no seguinte cenário:
Imagina que eu tenha apenas 10 customers persistidos na base de dados.
Se a chamada for assim `GET /v1/customers?offset=11&limit=10`.
Não existiria resgistros á serem retornados, pois não temos o décimo primeiro customer persistido.

De todo modo, isso foi só um detalhe.
No demais o código ficou muitooo bom, parabéns.